test/Stream/MemoryStreamReader.test.cpp

```diff
@@ -95,3 +97,36 @@ TEST_F(SimpleMemoryReader, SeekRelativeOutOfBoundsEndPreservesPosition) {
 	EXPECT_THROW(stream.SeekForward(6), std::runtime_error);
 	EXPECT_EQ(position, stream.Position());
 }
+
+TEST(MemoryReader, ReadNullTerminatedString)
+{
+	std::array<char, 5> buffer{ 'n', 'u', 'l', 'l', '\0' };
+	std::array<char, 5> result;
+
+	// Check null-terminated
+	Stream::MemoryReader reader(&buffer[0], 5);
+	std::string str;
+	
+	EXPECT_NO_THROW(str = reader.ReadNullTerminatedString());
+
+	// Copy to array
+	std::copy(str.begin(), str.end(), result.data());
+	
+	// Ensure null-terminator was not copied
+	EXPECT_NE(buffer, result);
```

Comment:
An `std::string` can contain embedded nulls. As such, you can actually just use a string compare to ensure it stopped before reading the null.

```cpp
assert(std::string("A", 1) != std::string("A\0", 2));
```

I believe the tests can be written without needing to use `result`.